### PR TITLE
pcap: Support jumbo frame

### PR DESCRIPTION
**Pull request by changlan**
This PR adds jumbo frame support to the PCAP driver, roughly following this DPDK patch: http://dpdk.org/ml/archives/dev/2015-February/014404.html.

### CODE REVIEW

File: core/drivers/pcap.cc [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #include "pcap.h"
 
+#include <algorithm>
```
Comment (sangjinhan) [inline core/drivers/pcap.cc]: Maybe I am missing something, but the code change doesn't seem to need these headers here...

File: core/drivers/pcap.cc [hunk 2]
```diff
@@ -47,27 +50,27 @@ int PCAPPort::RecvPackets(queue_t qid, bess::Packet** pkts, int cnt) {
...
+      m->set_next(bess::Packet::Alloc());
+      m = m->next();
+
+      // no headroom needed in chained mbufs
+      m->prepend(m->headroom());
+      m->set_data_len(0);
+      m->set_buffer(0);
+
+      copy_len = std::min(caplen, static_cast<int>(m->tailroom()));
+      bess::utils::CopyInlined(m->append(copy_len), packet, copy_len, true);
```
Comment (sangjinhan) [inline core/drivers/pcap.cc]: I think a non-inlined copy is good enough here... We already know that the packet is large, so tiny performance gain would not outweigh the increased code size.

File: core/drivers/pcap.cc [hunk 3]
```diff
@@ -47,27 +50,27 @@ int PCAPPort::RecvPackets(queue_t qid, bess::Packet** pkts, int cnt) {
...
-      break;
-#endif
+    int copy_len = std::min(caplen, static_cast<int>(sbuf->tailroom()));
+    bess::utils::CopyInlined(sbuf->append(copy_len), packet, copy_len, true);
+
+    packet += copy_len;
+    caplen -= copy_len;
+    bess::Packet* m = sbuf;
+
+    while (caplen > 0) {
```
Comment (sangjinhan) [inline core/drivers/pcap.cc]: We need to update the `nb_segs_` field if chained.

File: core/drivers/pcap.cc (hunk 1 above)
Comment (changlan) [inline core/drivers/pcap.cc]: `std::min` uses `<algorithm>`, and `std::string` is used too.

File: core/drivers/pcap.cc (hunk 3 above)
Comment (changlan) [inline core/drivers/pcap.cc]: Sure. Fixed.

**Comment by sangjinhan** [comment]
tumblr_m8im7ptqs81rdogh7o1_1280

**Pull request merged**